apps/scheduler/internal/timers/online.go

```diff
@@ -38,127 +38,105 @@ func NewOnlineUsers(ctx context.Context, services *types.Services) {
 					return
 				}
 
-				streamsWg := &sync.WaitGroup{}
-				for _, stream := range streams {
+				loParallel.ForEach(streams, func(stream *model.ChannelsStreams, _ int) {
 					if stream.Channel != nil && (!stream.Channel.IsEnabled || stream.Channel.IsBanned) {
-						continue
+						return
 					}
 
-					streamsWg.Add(1)
-
-					twitchClient, err := twitch.NewUserClient(stream.UserId, *services.Config, services.Grpc.Tokens)
+					twitchClient, err := twitch.NewUserClient(ctx, stream.UserId, *services.Config, services.Grpc.Tokens)
 					if err != nil {
 						zap.S().Error(err)
-						streamsWg.Done()
-						continue
+						return
 					}
 
-					go func(broadcasterId string) {
-						defer streamsWg.Done()
-						var chatters []helix.ChatChatter
-						cursor := ""
-						for {
-							req, err := twitchClient.GetChannelChatChatters(
-								&helix.GetChatChattersParams{
-									BroadcasterID: broadcasterId,
-									ModeratorID:   broadcasterId,
-									After:         cursor,
-								},
-							)
-							if err != nil {
-								zap.S().Error(err)
-							} else {
-								chatters = append(chatters, req.Data.Chatters...)
-								if req.Data.Pagination.Cursor == "" {
-									break
-								}
+					broadcasterId := stream.UserId
+					var chatters []helix.ChatChatter
+					cursor := ""
+					for {
+						req, err := twitchClient.GetChannelChatChatters(&helix.GetChatChattersParams{
+							BroadcasterID: broadcasterId,
+							ModeratorID:   broadcasterId,
+							After:         cursor, // TODO: cursor is not changing, should it?
```

Comment:
Yes, i forgot to change cursor after append on line 64

Reply:
fixed